ambari-server/src/main/java/org/apache/ambari/server/controller/metrics/timeline/cache/TimelineMetricCacheEntryFactory.java

```diff
@@ -319,4 +322,24 @@ private long getMillisecondsTime(long time) {
       return time;
     }
   }
+  @Override
+  public void delete(TimelineAppMetricCacheKey key) throws CacheWritingException {
+    // no need to implement.
+  }
+
+  @Override
+  public Map<TimelineAppMetricCacheKey, TimelineMetricsCacheValue> loadAll(Iterable<? extends TimelineAppMetricCacheKey> keys) throws BulkCacheLoadingException, Exception {
+//    no need to implement.
+    return null;
+  }
+
+  @Override
+  public void writeAll(Iterable<? extends Map.Entry<? extends TimelineAppMetricCacheKey, ? extends TimelineMetricsCacheValue>> entries) throws BulkCacheWritingException, Exception {
+//    no need to implement.
+  }
+
+  @Override
+  public void deleteAll(Iterable<? extends TimelineAppMetricCacheKey> keys) throws BulkCacheWritingException, Exception {
+//    no need to implement.
+  }
```

Comment:
If the code here is not specifically implemented, can it be removed？

Reply:
@JiaLiangC 
We are implementing CacheLoaderWriter interface, that's why we need to provide implementation of all its methods.